run_terminate_imagebuilder_gendo_old_environment.py

```diff
@@ -142,17 +180,28 @@ def run_terminate_image(name):
     rr = aws_cli.run(cmd)
     image_recipe_list = filter_imagebuilder_resource_arn_list(rr['imageRecipeSummaryList'])
 
-    cmd = ['logs', 'describe-log-groups']
-    cmd += ['--log-group-name-prefix', '/aws/imagebuilder/gendo']
-    rr = aws_cli.run(cmd, ignore_error=True)
-    imagebuilder_cw_log_list = rr['logGroups']
+    print_message('get component list')
+    cmd = ['imagebuilder', 'list-components']
+    rr = aws_cli.run(cmd)
+
+    component_list = filter_imagebuilder_resource_arn_list(rr['componentVersionList'])
+    timestamp_list = list()
+    for cc in component_list:
+        try:
+            found = re.search('-component-(.+?)/', cc).group(1)
+            if found:
+                timestamp_list.append(found)
+        except AttributeError:
+            pass
```

Comment:
@GoldieKMG // try-catch 는 가급적 1 라인으로 좁히는게 좋습니다. 그리고 안쓸수 있다면 안쓰는게 좋습니다. `re.search()`는 매칭이 실패할 경우 None을 리턴(https://docs.python.org/3/library/re.html#re.search) 하므로 아래와 같이 작성하여 try-catch를 회피 가능합니다.

```suggestion
    for cc in component_list:
        found = re.search('-component-(.+?)/', cc)
        if found:
           timestamp_list.append(found.group(1))
```

Reply:
@yoonjs2 // 네 회사 코드 컨벤션에서 말씀해주신대로 try catch는 필요할 때만 사용, 필요할 경우 1줄만 적용하는 부분 리마인드 하겠습니다 🙏